src/persist-client/src/impl/machine.rs

```diff
@@ -156,23 +156,39 @@ where
         &mut self,
         batch: &HollowBatch<T>,
         writer_id: &WriterId,
-    ) -> Result<
-        Result<Result<(SeqNo, Vec<FueledMergeReq<T>>), Upper<T>>, InvalidUsage<T>>,
-        Indeterminate,
-    > {
```

Comment:
I couldn’t come up with the hazard that would require keeping the `Indeterminate` in the signature. Probably just missing something, though.